frontend/src/features/admin-form/common/queries.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery, UseQueryResult } from 'react-query'
+import { useParams } from 'react-router-dom'
+
+import { AdminFormDto } from '~shared/types/form/form'
+
+import { getAdminFormView } from './AdminViewFormService'
+
+const adminFormKeys = {
+  base: ['adminForm'] as const,
+  id: (id: string) => ['adminForm', id] as const,
+}
+
+/**
+ * @precondition Must be wrapped in a Router as `useParam` is used.
+ */
+export const useAdminForm = (): UseQueryResult<AdminFormDto> => {
```

Comment:
should we type the error here?

```suggestion
export const useAdminForm = (): UseQueryResult<AdminFormDto, ErrorDto> => {
```

Reply:
Good point, but the error is already transformed in the interceptor; will update to the correct transformed error type